index.js

```diff
@@ -216,27 +231,29 @@ exports.validate = function (state, hmac_key, feed) {
   if(!state.feeds[feed] || !state.feeds[feed].queue.length) {
     return state
   }
-  var msg = state.feeds[feed].queue.pop()
+  var kvt = state.feeds[feed].queue.pop()
   state.queued -= 1
-  return exports.append(state, hmac_key, msg)
+  return exports.append(state, hmac_key, kvt.value)
```

Comment:
this rehashes the key. I got a bit confused reading append anyway, because it wasn't clear wether the msg arg was a kvt or a msg. I propose make an append_kvt method, and then append(..,msg) just applies toKeyValueTimestamp(msg) and calls `append_kvt`

OR, hmm, actually. most of the code in append should really be in validate, and append just calls queue then validate?

Reply:
You're right, well spottet. I'll check the first refactor and get back on the second one.